src/meta-srv/src/selector/lease_based.rs

```diff
@@ -28,24 +28,33 @@ impl Selector for LeaseBasedSelector {
     type Context = SelectorContext;
     type Output = Vec<Peer>;
 
-    async fn select(&self, ns: Namespace, ctx: &Self::Context) -> Result<Self::Output> {
-        // filter out the nodes out lease
-        let mut lease_kvs: Vec<_> =
-            lease::alive_datanodes(ns, &ctx.meta_peer_client, ctx.datanode_lease_secs)
-                .await?
-                .into_iter()
-                .collect();
-
-        lease_kvs.shuffle(&mut thread_rng());
-
-        let peers = lease_kvs
+    async fn select(
+        &self,
+        ns: Namespace,
+        ctx: &Self::Context,
+        opts: SelectorOptions,
+    ) -> Result<Self::Output> {
+        // 1. get alive datanodes.
+        let lease_kvs =
+            lease::alive_datanodes(ns, &ctx.meta_peer_client, ctx.datanode_lease_secs).await?;
+
+        // 2. compute weight array, but the weight of each item is the same.
+        let weight_array = lease_kvs
             .into_iter()
-            .map(|(k, v)| Peer {
-                id: k.node_id,
-                addr: v.node_addr,
+            .map(|(k, v)| WeightedItem {
+                item: Peer {
+                    id: k.node_id,
+                    addr: v.node_addr.clone(),
+                },
+                weight: 1,
+                reverse_weight: 1,
```

Comment:
Since the weight values are all the same, I think we should not apply the weight algorithm here, let's maintain the original logic of LeaseBased.